syntax = "proto3";

package syft.core.adp;
import "proto/core/common/common_object.proto";
import "proto/core/adp/entity.proto";

message IntermediateScalar {
  syft.core.common.UID id = 1;
  // poly
}

message BaseScalar {
  syft.core.common.UID id = 1;
  optional float min_val = 2;
  optional float value = 3;
  optional float max_val = 4;
  syft.core.adp.Entity entity = 5;
}

message IntermediateGammaScalar {
  syft.core.common.UID id = 1;
  // poly
}

message GammaScalar {
  syft.core.common.UID id = 1;
  optional float min_val = 2;
  optional float value = 3;
  optional float max_val = 4;
  syft.core.adp.Entity entity = 5;
  // poly
}

message IntermediatePhiScalar {
  syft.core.common.UID id = 1;
  syft.core.adp.Entity entity = 2;
  optional syft.core.adp.GammaScalar gamma = 3;
}

message PhiScalar {
  syft.core.common.UID id = 1;
  optional float min_val = 2;
  optional float value = 3;
  optional float max_val = 4;
  syft.core.adp.Entity entity = 5;
  optional syft.core.adp.GammaScalar gamma = 6;
  // poly
}
